packages/dds/merge-tree/src/attributionCollection.ts

```diff
@@ -53,18 +52,30 @@ export interface IAttributionCollection<T> {
     clone(): IAttributionCollection<T>;
 }
 
+
 export class AttributionCollection implements IAttributionCollection<AttributionKey> {
-    private readonly entries: RedBlackTree<number, number> = new RedBlackTree(compareNumbers);
+    private offsets: number[];
+    private seqs: number[];
 
     public constructor(baseEntry: number, private _length: number) {
-        this.entries.put(0, baseEntry);
+        this.offsets = [0];
```

Comment:
i feel like some of the bundle size increase can be offset by changing the constructor semantics of this. i kind of regret structuring it the way I did--making empty initialization the default with a singleton helper is probably more sensible and would simplify some of the logic in this class.

Reply:
reason why I think this should reduce the increase is because I had to add some additional logical checks to "avoid inserting for the first element" on repopulation and in a couple other places. The logic there should be simpler if we empty-initialize.

Reply:
I gave this a shot on Friday, and it doesn't help much. Just shifts the logic to other bits. If we do want to do this, it should be in a follow-up. Bundle size increase isn't drastic and the perf difference is, so PR still worth it imo.